### ISSUE TITLE
Development Ideas

### ISSUE THREAD

**Issue by friendly-bits** [lynxthecat/adblock-lean#59]
There are lots of ideas discussed in the forum thread, so let's keep track of the ones which seem to be feasible here.
(this is open for dispute if anyone thinks that some idea is no good or should be changed)
**I'd suggest to keep this issue as a list of ideas, and if you want to discuss an individual idea then open a separate issue for it.**

- Change config options `report_failure`, `report_success` to `custom_script`, run the functions `report_success()`, `report_failure()` from the custom script on exit rather than doing eval on the variable contents
- Make current session log available to the luci RPC script and to the custom script via a variable. Could be implemented either by appending each new log entry to a temp file, or as suggested by @dave14305 , by adding the current PID to the log entries and then using `logread -e "adblock-lean[$$]"` to retrieve them. I like the simple temp file solution, if anyone strongly prefers the logread solution, let me know.
- Add a custom command to retrieve last session log (simplest implementation would be keeping the last session log file and reading it out on-demand, but that would be only possible if the above idea is implemented via a temp file, otherwise we would need to make a marker log entry at session start and use it to separate the sessions)
- Unify all the possible part fail options into 1 option, e.g. part_fail_action. This would cover all possible fail conditions, while making the logic a bit simpler, and making it easier for the user to understand the config options.

**Comment by rickparrish** [comment]
Don't think I have enough input to create separate issues, so just adding my $0.02 here (which is mostly just saying "I'm all for these changes")

- I love the custom script idea.  I'm not confident that I could escape things correctly to enter report_failure or report_success via a textarea on the luci app, but think I could figure it out via a custom script.
- And the session log will be very useful for the luci app.  Yesterday I discovered RPC calls timeout after 30 seconds, and I haven't been able to figure out how to extend that.  That means the Start button can't use RPC, so won't have access to any of the luci_* variables, so being able to get the session log from a temp file would be good.
- Would the "retrieve last session" command be for the luci app, or have other purpose too?  If it's just for the luci app, and if the session log is stored to disk, then the luci app can read the file directly and I'd probably do that instead of use the command.  If it's not stored to disk, then that's where the command could be convenient.

**Comment by lynxthecat** [comment]
How about local blocklist supports IPs?

**Comment by lynxthecat** [comment]
Nothing comes to mind, but it does seem like an extra feature to advertise. There was mention of this in the forum [here](https://forum.openwrt.org/t/does-dnsmasq-full-supports-regex-entry/229445/2).

**Comment by friendly-bits** [comment]
While dnsmasq does support this, IMO blocking IP addresses is more efficient when done in the firewall (i.e. via nftables rules). If there really is a use case, we could either go the lazy route and just use dnsmasq, or add another module to adblock-lean which would create nft sets and firewall rules for those sets. The latter is obviously significantly more work, but this is how it's supposed to be done IMO. Generally, I feel that this goes beyond the core purpose of an adblocker, but I'm open for discussion.

**Comment by lynxthecat** [comment]
Yes I see your point. From symmetry perspective though we already support remote IP lists don’t we? So there is that. Maybe this is something to keep in mind in case a use case comes up in the future. Or an especially rainy day/
